### PR TITLE
PAS-202 | Add OpenAPI to onboarding page + prettify with icons

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Ticket
- Closes [PAS-202](https://bitwarden.atlassian.net/browse/PAS-202)

### Description

- Adds the generated OpenAPI documentation to `/app/{appid}/onboarding/get-started` page.
- Add icons to every front-end and back-end SDK
- Add missing Android SDK.
- For self-hosted environments, the REST documentation will redirect to `/swagger/index.html` rather than the documentation at `https://docs.passwordless.dev/guide/api-documentation.html`.

### Shape

- Page migrated to Blazor
- Download handler for onboarding keys is properly protected by the `HasAppHandler` authorization policy. 
- Depends on: [https://github.com/bitwarden/passwordless-docs/pull/104](https://github.com/bitwarden/passwordless-docs/pull/104). The documentation pull request does require the API to be deployed first for the CORS headers to be properly applied to the swagger schema. So this pull request can be safely merged in the mean-time.
- Remove Vue.js.

### Screenshots

A good reminder of what we had before, credits go to my bad design skills.

image

After:

https://github.com/bitwarden/passwordless-server/assets/1045327/3bb154b2-0839-45f8-b758-41dcda92b672

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- Tested locally

I did the following to ensure that my changes do not introduce security vulnerabilities:
- Downloading the onboarding keys has been tested to be properly secured by the `HasAppHandler` authorization policy.
- Verified that impersonation of the migrated onboarding page is not vulnerable to impersonation.

[PAS-202]: https://bitwarden.atlassian.net/browse/PAS-202?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/AdminConsole/Components/Shared/Cards/LinkCard.razor [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<a href="@Model.Link" target="_blank">
+    <div class="relative flex rounded-lg border bg-white p-4 shadow-sm focus:outline-none">
+        <div class="flex-auto flex flex-col space-y-1">
+            <span class="block text-sm font-medium text-gray-900">@Model.Title</span>
+            <span class="flex items-center text-sm text-gray-500">@Model.Description</span>
+        </div>
+        
+        @if (Model.Image != null)
+        {
+            <DynamicComponent Type="@Model.Image" Parameters="@(new Dictionary<string, object> { { "class", "flex-none w-10 h-10" } })"/>
```
Comment (jonashendrickx) [inline src/AdminConsole/Components/Shared/Cards/LinkCard.razor:10]: This just passes the type of the component, and allows us to inject any component dynamically. This is quite nice.

File: src/AdminConsole/Endpoints/ApplicationEndpoints.cs [hunk 2]
```diff
@@ -13,6 +17,10 @@ public static IEndpointRouteBuilder MapApplicationEndpoints(this IEndpointRouteB
 
         // The Blazor SSR sample uses the same solution for signing out, but we do not want to use all the endpoints.
         group.MapGet("/{appId}/Available", IsAppIdAvailableAsync).RequireAuthorization();
+
+        group.MapGet("/{appId}/DownloadOnboarding", DownloadOnboardingAsync)
+            .RequireAuthorization(CustomPolicy.HasAppRole);
```
Comment (jonashendrickx) [inline src/AdminConsole/Endpoints/ApplicationEndpoints.cs:22]: The `HasAppRole` policy is the same one that checks if we're allowed to access an application.

File: src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+@inject IOptionsSnapshot<PasswordlessManagementOptions> PasswordlessOptions
+
+@if (_isInitialized)
+{
+    <Page>
+        <Panel Header="1. Your API Keys">
+            <div class="flex flex-col space-y-4 mt-2 max-w-xl text-sm text-gray-500">
+                <p>Let's get set up using Passwordless.dev. Please feel free to read all the <a target="_blank" href="https://docs.passwordless.dev" class="link-blue">documentation</a>.</p>
+            
+                <p>Here are your API keys. You will use these to access the Passwordless api. The Api Key is used client side and is safe to share, while the ApiSecret is used on your backend and should never be exposed.</p>
```
Comment (Tyrrrz) [inline src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor:19]: More consistent spacing and capitalization:

```suggestion
                <p>Here are your API keys. You will use these to access the Passwordless api. The API Key is used client-side and is safe to share, while the API Secret is used on your backend and should never be exposed.</p>
```

File: src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                    <form id="downloadForm" method="get" action="/Applications/@AppId/DownloadOnboarding">
+                        <button type="submit" class="btn-primary btn-icon"><DownloadIcon/> Download</button>
+                    </form>
+                </div>
+            </div>
+        </Panel>
+
+        <Panel Header="2. Backend language">
+            <div class="mt-2 max-w-xl text-sm text-gray-500">
+                <p>Passwordless.dev works with <strong>all backend languages</strong> using a normal REST API. But for some languages, we've made official packages, sdks and examples.</p>
```
Comment (Tyrrrz) [inline src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor:42]:
```suggestion
                <p>Passwordless.dev works with <strong>all backend languages</strong> using a normal REST API. But for some languages, we provide official packages, SDKs and examples.</p>
```

File: src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+            <PanelFooter>
+                <a target="_blank" href="https://docs.passwordless.dev/guide/backend/" class="link-blue">View the backend examples <span aria-hidden="true">&rarr;</span></a>
+                <a target="_blank" href="https://docs.passwordless.dev/guide/api-documentation.html" class="link-blue">Read the API docs <span aria-hidden="true">&rarr;</span></a>
+            </PanelFooter>
+        </Panel>
+
+        <Panel Header="3. Frontend framework">
+            <div class="mt-2 max-w-xl text-sm text-gray-500">
+                <p>Passwordless.dev uses a little bit of javascript and html. It works regardless if you're using a framework or not! For some frameworks, we've made official packages, sdks and examples.</p>
```
Comment (Tyrrrz) [inline src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor:60]:
```suggestion
                <p>Passwordless.dev uses a little bit of JavaScript and HTML. It works regardless if you're using a framework or not! For some frameworks, we provide official packages, SDKs and examples.</p>
```

File: src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        new("Python", "Docs + Examples", typeof(PythonIcon), new Uri("https://docs.passwordless.dev/guide/backend/python3.html"))
+    };
+
+    public IEnumerable<LinkCard.LinkCardModel> FrontendCards => new LinkCard.LinkCardModel[]
+    {
+        new("Android", "Docs + Examples", typeof(AndroidIcon), new Uri("https://docs.passwordless.dev/guide/frontend/android.html")),
+        new("ASP.NET Identity", "Docs + Examples", typeof(DotNetIcon), new Uri("https://docs.passwordless.dev/guide/frontend/aspnet.html")),
+        new("Javascript", "Docs + Examples", typeof(JavascriptIcon), new Uri("https://docs.passwordless.dev/guide/frontend/javascript.html")),
+        new("React", "Docs + Examples", typeof(ReactIcon), new Uri("https://docs.passwordless.dev/guide/frontend/react.html"))
+    };
```
Comment (Tyrrrz) [inline src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor.cs:24]: Avoid recreating the arrays every time the property is accessed:

```suggestion
    public IReadOnlyList<LinkCard.LinkCardModel> BackendCards { get; } = new LinkCard.LinkCardModel[]
    {
        new("REST API", "Use our simple REST API", typeof(OpenApiIcon), new Uri(Configuration.IsSelfHosted() ? $"{PasswordlessOptions.Value.ApiUrl}/swagger/index.html" : "https://docs.passwordless.dev/guide/api-documentation.html")),
        new(".NET", "Docs + Examples", typeof(DotNetIcon), new Uri("https://docs.passwordless.dev/guide/backend/dotnet.html")),
        new("Java", "Docs + Examples", typeof(JavaIcon), new Uri("https://docs.passwordless.dev/guide/backend/java.html")),
        new("Node.js", "Docs + Examples", typeof(NodeJsIcon), new Uri("https://docs.passwordless.dev/guide/backend/nodejs.html")),
        new("Python", "Docs + Examples", typeof(PythonIcon), new Uri("https://docs.passwordless.dev/guide/backend/python3.html"))
    };

    public IReadOnlyList<LinkCard.LinkCardModel> FrontendCards { get; } = new LinkCard.LinkCardModel[]
    {
        new("Android", "Docs + Examples", typeof(AndroidIcon), new Uri("https://docs.passwordless.dev/guide/frontend/android.html")),
        new("ASP.NET Identity", "Docs + Examples", typeof(DotNetIcon), new Uri("https://docs.passwordless.dev/guide/frontend/aspnet.html")),
        new("Javascript", "Docs + Examples", typeof(JavascriptIcon), new Uri("https://docs.passwordless.dev/guide/frontend/javascript.html")),
        new("React", "Docs + Examples", typeof(ReactIcon), new Uri("https://docs.passwordless.dev/guide/frontend/react.html"))
    };
```

File: src/AdminConsole/Endpoints/ApplicationEndpoints.cs [hunk 7]
```diff
@@ -33,4 +41,25 @@ public static IEndpointRouteBuilder MapApplicationEndpoints(this IEndpointRouteB
...
+            return NotFound();
+        }
+
+        var contentBuilder = new StringBuilder();
+        contentBuilder.AppendLine($"API Url: {options.Value.ApiUrl}");
+        contentBuilder.AppendLine($"Public ApiKey: {onboarding.ApiKey}");
+        contentBuilder.AppendLine($"Private ApiSecret: {onboarding.ApiSecret}");
+
+        var byteArray = Encoding.UTF8.GetBytes(contentBuilder.ToString());
+        return File(byteArray, "application/octet-stream", $"passwordless-apikeys-{appId}.txt");
```
Comment (Tyrrrz) [inline src/AdminConsole/Endpoints/ApplicationEndpoints.cs:63]: Could be simplified with slightly fewer allocations:

```suggestion
        using var memory = new MemoryStream();
        using var writer = new StreamWriter(memory);
        writer.WriteLine($"API Url: {options.Value.ApiUrl}");
        writer.WriteLine($"Public ApiKey: {onboarding.ApiKey}");
        writer.WriteLine($"Private ApiSecret: {onboarding.ApiSecret}");

        return File(memory.ToArray(), "application/octet-stream", $"passwordless-apikeys-{appId}.txt");
```

File: src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor.cs (hunk 6 above)
Comment (jonashendrickx) [reply src/AdminConsole/Components/Pages/App/Onboarding/GetStarted.razor.cs]: Configuration would be accessed from a static context, wouldn't be possible.

**Review by Tyrrrz — APPROVED** [review]

**Pull request merged**